src/poll.rs

```diff
@@ -47,3 +50,19 @@ pub fn poll(fds: &mut [PollFd], timeout: libc::c_int) -> Result<libc::c_int> {
 
     Errno::result(res)
 }
+
+pub fn ppoll(fds: &mut [PollFd], timeout: Duration) -> Result<libc::c_int> {
```

Comment:
we should accept a sigmask here, too. @fiveop @chaosagent could you suggest how to represent it in nix?